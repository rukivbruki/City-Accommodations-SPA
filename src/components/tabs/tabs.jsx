import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {ActionCreator} from "../../reducer/data/data";
import TabsLink from "../tabs-link/tabs-link";
import {getCities} from "../../reducer/data/selector";

const Tabs = (props) => {
  const {
    currentCity,
    activeElement,
    changeCurrentCity,
    cities,
    onSelect,
  } = props;

  return (
    <div className="tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.map((city, index) =>
            <TabsLink
              key={`${index}-${city}`}
              city={city}
              onSelect={onSelect}
              activeElement={activeElement || currentCity}
              changeCurrentCity={changeCurrentCity}
            />)}
        </ul>
      </section>
    </div>
  );
};

Tabs.propTypes = {
  activeElement: PropTypes.string,
  changeCurrentCity: PropTypes.func.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentCity: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  cities: getCities(state)
});

const mapDispatchToProps = (dispatch) => ({
  changeCurrentCity: (city) => dispatch(ActionCreator.changeCity(city))
});

export {Tabs};
export default connect(mapStateToProps, mapDispatchToProps)(Tabs);
