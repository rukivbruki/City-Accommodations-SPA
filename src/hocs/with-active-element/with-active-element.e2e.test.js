import React from "react";
import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
import withActiveElement from "./with-active-element";
import {Tabs} from "../../components/tabs/tabs";

Enzyme.configure({adapter: new Adapter()});

describe(`PlacesListWrapped works correct`, () => {
  const TabsWrapped = withActiveElement(Tabs);
  const changeCurrentCity = jest.fn();
  const cities = [`Paris`, `Kyiv`];

  const wrappedComponent = mount(
      <TabsWrapped
        changeCurrentCity={changeCurrentCity}
        cities={cities}
        currentCity={`Amsterdam`}
      />
  );

  const firstTab = wrappedComponent.find(`.locations__item-link`).first();


  it(`PlacesListWrapped should render correctly with offers`, () => {
    expect(toJson(wrappedComponent)).toMatchSnapshot();
  });

  it(`PlacesListWrapped must have initial state === null`, () => {
    expect(wrappedComponent.state().activeElement).toEqual(null);
  });

  it(`PlacesListWrapped change state on mouse hover first offer`, () => {
    firstTab.simulate(`click`);

    expect(wrappedComponent.state().activeElement).toEqual(`Paris`);
  });
});
