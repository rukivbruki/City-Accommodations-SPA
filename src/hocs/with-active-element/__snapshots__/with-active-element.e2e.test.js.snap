// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesListWrapped works correct PlacesListWrapped should render correctly with offers 1`] = `
<WithActiveElement
  changeCurrentCity={[MockFunction]}
  cities={
    Array [
      "Paris",
      "Kyiv",
    ]
  }
  currentCity="Amsterdam"
>
  <Tabs
    activeElement={null}
    changeCurrentCity={[MockFunction]}
    cities={
      Array [
        "Paris",
        "Kyiv",
      ]
    }
    currentCity="Amsterdam"
    onSelect={[Function]}
  >
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <TabsLink
            activeElement="Amsterdam"
            changeCurrentCity={[MockFunction]}
            city="Paris"
            key="0-Paris"
            onSelect={[Function]}
          >
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item "
                href="#"
                onClick={[Function]}
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
          </TabsLink>
          <TabsLink
            activeElement="Amsterdam"
            changeCurrentCity={[MockFunction]}
            city="Kyiv"
            key="1-Kyiv"
            onSelect={[Function]}
          >
            <li
              className="locations__item"
            >
              <a
                className="locations__item-link tabs__item "
                href="#"
                onClick={[Function]}
              >
                <span>
                  Kyiv
                </span>
              </a>
            </li>
          </TabsLink>
        </ul>
      </section>
    </div>
  </Tabs>
</WithActiveElement>
`;
